### PR TITLE
Ophase Categories

### LINKED ISSUE #56: Link Jobs with Groupcategories
Some jobs are only needed if there are groups of one or multiple specific categories. It would be nice to hide these jobs if none of those categories is active.

This consists of two parts:

- [x] Linking the models of job and group category with a ManyToMany relation (while using None as default for a job belonging to all categories without the need to explicitly model this)
- [x]  Hiding all jobs that are not universal and do not belong to any of the active categories

### PR DESCRIPTION
**Pull request by bhaettasch**
Introducing and using different categories for Ophase to e.g. allow a Ophase with only Master tutor groups (already possible without these changes) but also only Jobs for a Master Ophase (using changes from this branch/pull request).

Work in progress. Feel free to comment anyway.

This aims to resolve #56

### CODE REVIEW

**Comment by bhaettasch** [comment]
Not sure why the tests are failing with

> ValueError: Related model 'ophasebase.OphaseCategory' cannot be resolved

suddenly. Has anyone an idea? Maybe @ckleemann?

Also I am a little bit annoyed by those fake "new issues" of codeclimate from lots of files I did not even touch.

**Comment by bhaettasch** [comment]
I was able to track down the error -- it was caused by migrations depending on a specific run order that was used locally during incremental migrations but not when populating a new database. No need to check anymore, @ckleemann.

**Comment by bhaettasch** [comment]
Contrary to the commit message of	bd5c9e2, this commit cannot be cherry picked since a migration is contained (and the dependencies would not be fulfilled when picking).

**Comment by schroedingersZombie** [comment]
We would like to deploy this features asap so that the hompage of pyOphase can be updatet to the next Ophase.

File: ophasebase/tests.py [hunk 1]
```diff
@@ -13,10 +13,10 @@ class OphaseIsActive(TestCase):
 
     def test_only_one_is_active(self):
         # create Ophase - should be active
-        o1 = Ophase.objects.create(start_date=date(2014, 4, 7), end_date=date(2014, 4, 11), is_active=True)
+        o1 = Ophase.objects.create(name="Testophase 1", is_active=True)
```
Comment (jlauinger) [inline ophasebase/tests.py]: It appears to me as if this only tests the Ophase instance itself, not any OphaseCategories. You might want to add some OphaseActiveCategories to o1.

Comment (bhaettasch) [inline ophasebase/tests.py]: I created issue #131 to make sure this does not fall into oblivion.

**Comment by bhaettasch** [comment]
Reports are working, I just checked it.

Squashing is probably not a good idea since I had to edit the migrations manually to make sure migrating works for existing databases as well as empty ones.

The translations are already done on the main branch (with #128) and will be complete after merging.

**Comment by bhaettasch** [comment]
From my point of view, the essential work is done here and this branch can be merged. All further improvements and usages are enhancements and can be added when needed.

**Pull request merged**